lib/files.go

```diff
@@ -49,7 +45,7 @@ func CheckFileExist(file string) bool {
 
 // Unzip will decompress a zip archive, moving all files and folders
 // within the zip file (parameter 1) to an output directory (parameter 2).
-func Unzip(src string, dest string) ([]string, error) {
+func Unzip(src string, dest string, fileToUnzip string) ([]string, error) {
```

Comment:
This breaks a public method *and* it's being used!
https://github.com/hazelops/ize/blob/97d7c0e48f116054b7493dfcaa15bd0460bae016/internal/terraform/local.go#L274

Reply:
I'm afraid I'm quite new to golang.. though I've dealt with default argument values before - I can't quite remember how (I had a feeling it was using a (without knowing a better word for it) expansion technique (`(params ...string)`).
If this is the "best" way, I guess it would need to do something like:

```
func Unzip(src string, dest, string, fileToUnzipSlice ...string) {
  fileToUnzip := "terraform"
  if len(fileToUnzipSlice) > 0 {
    fileToUnzip = fileToUnzipSlice[0]
  }
}
```
?

Reply:
> I'm afraid I'm quite new to golang..

Whereas I'm a way not a Golang guy at all 🤣 But, yeah, using ellipsis looks easiest way.
I'd only say let's check the number of args to prevent consumers passing improper count of args by an accident or a human mistake. Plus an explanatory comment. Like this:
```go
// Make 3rd arg optional for backward compatibility
func Unzip(src string, dest string, fileToUnzipSlice ...string) {
  fileToUnzip := "terraform"
  if len(fileToUnzipSlice) > 1 {
    log.Fatal("Too many args")
  } else if len(fileToUnzipSlice) == 1 {
    fileToUnzip = fileToUnzipSlice[0]
  }
}
```

Reply:
I also assume `fileToUnzip := "terraform"` would refer not `terraform` verbatim, but would be a ref to a var that holds default product's executable filename.

Reply:
Done :)